stm32_base/hal/timer_stm32/timer_device_stm32.c

```diff
@@ -271,21 +299,11 @@ bool timer_device_init(const size_t channel, const uint32_t speed,
                 return false;
 
         struct config *c = &g_config[channel];
-        c->prescaler = speed_to_prescaler(speed);
+        c->prescaler = speed_to_prescaler(channel, speed);
```

Comment:
what's the risk of setting the prescaler to 0 if we've passed in an invalid channel? The timer may misbehave?